src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -934,7 +901,7 @@ void SpatialPooler::inhibitColumnsLocal_(const vector<Real> &overlaps,
 
 
 bool SpatialPooler::isUpdateRound_() const {
-  return (iterationNum_ % updatePeriod_) == 0;
+  return (rng_.getReal64() < 1.0/updatePeriod_); //approx every updatePeriod steps
```

Comment:
this is one place where `iterationNum_` was needed, I have replaced with probability, which I think is nicer approach.